File: Python/test/test_leetcode_007.py
from src import leetcode_007


class TestClass001(object):
    def test_solution_1(self):
        assert leetcode_007.Solution().reverse(123) == 321

    def test_solution_2(self):
        assert leetcode_007.Solution().reverse(-123) == -321

    def test_solution_3(self):
        assert leetcode_007.Solution().reverse(120) == 21

File: C/test/test_leetcode_007.c
#include "../src/leetcode_007.c"
#include <assert.h>
#include <stdio.h>

#define sizeof_array(array) (sizeof(array) / sizeof(array[0]))

void run_tests() {
  assert(reverse(123) == 321);
  assert(reverse(-123) == -321);
  assert(reverse(120) == 21);

  printf("LEETCODE_007.C | All tests passed.\n");
}

int main() {
  run_tests();
  return 0;
}
